### DISCUSSION TITLE
overwrite paste

### DISCUSSION THREAD

**Discussion by vincentaxhe** [helix-editor/helix#11718] [Q&A] [answered]
yank something, select other something, paste but not what I want to overwrite selected region, if I press 'd' to delete region then yanked buffer is replaced, how do I deal with this situation?

**Comment by rahil627** [comment]
c for change

**Reply by vincentaxhe** [reply]
nono it's not as simple as that, I want selection area replaced by buffer,helix just ignore selection and paste, any 'd' or 'c' will just waste the desired buffer. I want helix be smarter when paste on a selected area, clear the area and paste.Also it's the normal text editor will do and neovim able to do it.

**Reply by gabydd** [reply, marked as answer]
I think you might want ``R`` for register replace or ``<space>R`` for system clipboard replace

**Reply by vincentaxhe** [reply]
yes, I drop out half way helix tutor, Helix thought it as independent 'replace' action.

**Reply by rahil627** [reply]
ah gotcha. your writing was bit difficult to understand. maybe translated? glad someone else answered tho! 😄

**Reply by vincentaxhe** [reply]
no, I'm Chinese, translation makes me feel like walking with other's underware. I did not polish the words since I feel discussion ground to be a little free and easy place compared to issue. Glad you answer it, make me clarify my question.

**Reply by rahil627** [reply]
hmmm, i'd like to feel like walking in another person's underwear... maybe i need to try translating! 🙊
